src_c/surface.c

```diff
@@ -2097,6 +2101,215 @@ surf_blits(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
+            goto on_error;
+        }
+    }
+
+    blitsequence = args[0];
+
+    if (!dest) {
+        errornum = BLITS_ERR_DISPLAY_SURF_QUIT;
+        goto on_error;
+    }
+
+    /* List or Tuple */
+    if (PyList_Check(blitsequence) || PyTuple_Check(blitsequence)) {
+        f_blitsequence = PySequence_Fast_ITEMS(blitsequence);
+        sequencelength = PySequence_Fast_GET_SIZE(blitsequence);
+        for (seq_counter = 0; seq_counter < sequencelength; seq_counter++) {
+            item = f_blitsequence[seq_counter];
+            if (PyTuple_Check(item)) {
+                itemlength = PySequence_Fast_GET_SIZE(item);
+                if (itemlength != 2) {
+                    errornum = FBLITS_ERR_TUPLE_REQUIRED;
+                    goto on_error;
+                }
+            }
+            else {
+                errornum = FBLITS_ERR_TUPLE_REQUIRED;
+                goto on_error;
+            }
+
+            /* (Surface, dest)
+             * using PyTuple_GET_ITEM for better perf
+             * because the docs say it must be a tuple */
+            srcobject = PySequence_Fast_GET_ITEM(item, 0);
+            argpos = PySequence_Fast_GET_ITEM(item, 1);
+
+            if (!pgSurface_Check(srcobject)) {
+                errornum = BLITS_ERR_SOURCE_NOT_SURFACE;
+                goto on_error;
+            }
+
+            src = pgSurface_AsSurface(srcobject);
+            if (!src) {
+                errornum = BLITS_ERR_SEQUENCE_SURF;
+                goto on_error;
+            }
+
+            if (pg_TwoIntsFromObj(argpos, &(dest_rect.x), &(dest_rect.y))) {
+            }
+            else if ((src_rect = pgRect_FromObject(argpos, &temp))) {
+                dest_rect.x = src_rect->x;
+                dest_rect.y = src_rect->y;
+            }
+            else {
+                errornum = BLITS_ERR_INVALID_DESTINATION;
+                goto on_error;
+            }
+
+            dest_rect.w = src->w;
+            dest_rect.h = src->h;
+
+            result = pgSurface_Blit(self, (pgSurfaceObject *)srcobject,
+                                    &dest_rect, NULL, flags_numeric);
+
+            if (result != 0) {
+                errornum = BLITS_ERR_BLIT_FAIL;
+                goto on_error;
+            }
+        }
+    }
+    /* Generator */
+    else if (PyIter_Check(blitsequence)) {
+        while ((item = PyIter_Next(blitsequence))) {
+            if (PySequence_Check(item)) {
+                itemlength = PySequence_Size(item);
+                if (itemlength != 2) {
+                    errornum = FBLITS_ERR_TUPLE_REQUIRED;
+                    goto on_error;
+                }
+            }
+            else {
+                errornum = FBLITS_ERR_TUPLE_REQUIRED;
+                goto on_error;
+            }
+
+            /* (Surface, dest) */
+            srcobject = PySequence_ITEM(item, 0);
+            argpos = PySequence_ITEM(item, 1);
+
+            Py_DECREF(item);
```

Comment:
As I understand `PySequence_ITEM`, "we" receive are now responsible for cleanup (`Py_DECREF`) of these parameters.

But even if I am wrong here and we get a borrowed reference, then we have another problem - namely that we are dereferencing the tuple containing these two items, which might invalidate the references we just pulled out of it.

Either way, I suspect something is wrong here.